core/client/fs/src/main/java/alluxio/client/file/BaseFileSystem.java

```diff
@@ -462,4 +480,45 @@ public void unmount(AlluxioURI path, UnmountOptions options)
       mFileSystemContext.releaseMasterClient(masterClient);
     }
   }
+
+  /**
+   * Alluxio URIs in the Filesystem don't need a scheme or host. Warn the user and throw an
+   * exception if necessary.
+   */
+  protected static void UriCheck(AlluxioURI uri) {
+    boolean logWarning = false;
+    if (uri.hasScheme()) {
+      if (uri.getScheme().compareTo("alluxio") == 0
+          || uri.getScheme().compareTo("alluxio-ft") == 0) {
+        logWarning = true;
+      } else {
+        // Throw Exception
+        throw new IllegalArgumentException(
+            "Valid scheme values in filesystem URI operations are \"alluxio\" and \"alluxio-ft\"."
+                + " Note that the scheme is not required for filesystem operations");
+      }
+    }
+
+    if (uri.hasAuthority()) {
+      logWarning = true;
+      /* Even if we choose to log the warning, check if the Configuration host matches what the
+       * user passes. If not, throw an exception letting the user know they don't match.
+       */
+      String host = Configuration.get(PropertyKey.MASTER_HOSTNAME);
+      String port = Configuration.get(PropertyKey.MASTER_RPC_PORT);
+      String authString = String.format("%s:%s", host, port);
+      Authority trueAuthority = Authority.fromString(authString);
```

Comment:
it's inefficient to build a string and then parse it. If we modify `SingleMasterAuthority` to have a `String, int` constructor, we can build a `SingleMasterAuthority` directly and compare to `uri.getAuthority()`. We should update `SingleMasterAuthority` to get rid of the superfluous `mAuthority` field

Reply:
So to make sure I understand, I should modify the `SingleMasterAuthority` class and get change the constructor so that it is `(String host, int port)` and remove the `mAuthority` field all together. This means the `toString()` method of the class needs to be changed as well as anywhere else that calls the SingleMasterAuthority` constructor (There's only 1 usage). And in my code I should just create a new SingleMasterAuthority class

Reply:
that's exactly what I meant 👍